test/e2e/node/stress_probe.go

```diff
@@ -0,0 +1,196 @@
+	pod.Spec.Containers[0].Name = "agnhost-container-0"
+	pod.Spec.Containers = append(pod.Spec.Containers, make([]v1.Container, numContainers-1)...)
+	for idx := 0; idx < numContainers; idx++ {
+		name := "agnhost-container-" + strconv.Itoa(idx+1)
+		instance := *pod.Spec.Containers[0].DeepCopy()
+		pod.Spec.Containers[idx] = newContainer(name, probeType, idx, instance)
+	}
+	return pod
+}
+
+func newContainer(name, probeType string, containerIdx int, instance v1.Container) v1.Container {
+	newContainerPort := containerIdx + containerPort + 1
+	instance.Name = name
+	instance.Ports = []v1.ContainerPort{{ContainerPort: int32(newContainerPort)}}
+
+	switch probeType {
+	case "tcp":
+		instance.LivenessProbe.TCPSocket.Port = intstr.FromInt(newContainerPort)
+		instance.Args[3] = strconv.Itoa(newContainerPort)
+	case "http":
+		instance.LivenessProbe.HTTPGet.Port = intstr.FromInt(newContainerPort)
+		instance.Args[3] = strconv.Itoa(newContainerPort)
+	case "grpc":
+		instance.LivenessProbe.GRPC.Port = int32(newContainerPort)
+		instance.Args[2] = strconv.Itoa(newContainerPort)
+		instance.Args[4] = strconv.Itoa(newContainerPort + containerPort)
+	}
+
+	return instance
+}
+
+func httpGetHandler(path string, port int) v1.ProbeHandler {
+	return v1.ProbeHandler{
+		HTTPGet: &v1.HTTPGetAction{
+			Path: path,
+			Port: intstr.FromInt(port),
+		},
+	}
+}
+
+func tcpSocketHandler(port int) v1.ProbeHandler {
+	return v1.ProbeHandler{
+		TCPSocket: &v1.TCPSocketAction{
+			Port: intstr.FromInt(port),
+		},
+	}
+}
+
+func getRestartCount(p *v1.Pod) int {
+	count := 0
+	for _, containerStatus := range p.Status.ContainerStatuses {
+		count += int(containerStatus.RestartCount)
+	}
+	return count
+}
+
+// runLivenessTest verifies the number of restarts.
+func runLivenessTest(ctx context.Context, f *framework.Framework, pod *v1.Pod) {
+	podClient := e2epod.NewPodClient(f)
+	ns := f.Namespace.Name
+	gomega.Expect(pod.Spec.Containers).NotTo(gomega.BeEmpty())
+	// At the end of the test, clean up by removing the pod.
+	ginkgo.DeferCleanup(func(ctx context.Context) error {
+		ginkgo.By("deleting the pod")
+		return podClient.Delete(ctx, pod.Name, *metav1.NewDeleteOptions(0))
+	})
+	ginkgo.By(fmt.Sprintf("Creating pod %s in namespace %s", pod.Name, ns))
+	podClient.Create(ctx, pod)
+
+	// Wait until the pod is not pending. (Here we need to check for something other than
+	// 'Pending' other than checking for 'Running', since when failures occur, we go to
+	// 'Terminated' which can cause indefinite blocking.)
+	framework.ExpectNoError(e2epod.WaitForPodNotPending(ctx, f.ClientSet, ns, pod.Name),
+		fmt.Sprintf("starting pod %s in namespace %s", pod.Name, ns))
+	framework.Logf("Started pod %s in namespace %s", pod.Name, ns)
+
+	// Check the pod's current state and verify that restartCount is present.
+	ginkgo.By("checking the pod's current state and verifying that restartCount is present")
+	pod, err := podClient.Get(ctx, pod.Name, metav1.GetOptions{})
```

Comment:
the problem with the probes is that they accumulate connections, we should check X times during Y minutes/seconds, at least 2 mins

Reply:
yeah, I just removed that part, I will add it back